sdk/nodejs/project.json

```diff
@@ -1,70 +1,87 @@
-                "passWithNoTests": false,
-                "codeCoverage": true
-            },
-            "dependsOn": [
-                {
-                    "target": "build"
-                },
-                {
-                    "target": "build:json",
-                    "dependencies": true
-                }
-            ]
-        },
-        "build": {
-            "executor": "@nx/js:tsc",
-            "outputs": ["{options.outputPath}"],
-            "options": {
-                "outputPath": "dist/sdk/nodejs",
-                "tsConfig": "sdk/nodejs/tsconfig.lib.json",
-                "packageJson": "sdk/nodejs/package.json",
-                "buildableProjectDepsInPackageJsonType": "dependencies",
-                "main": "sdk/nodejs/src/index.ts",
-                "assets": ["sdk/nodejs/*.md"]
-            },
-            "dependsOn": [
-                {
-                    "target": "build",
-                    "dependencies": true
-                }
-            ]
+  "name": "nodejs",
+  "$schema": "../../node_modules/nx/schemas/project-schema.json",
+  "sourceRoot": "sdk/nodejs/src",
+  "implicitDependencies": [
+    "shared-bucketing-as",
+    "shared-bucketing-test-data"
+  ],
+  "targets": {
+    "check-types": {
+      "executor": "nx:run-commands",
+      "options": {
+        "command": "yarn run -T tsc -b --incremental",
+        "cwd": "sdk/nodejs"
+      }
+    },
+    "lint": {
+      "executor": "@nx/linter:eslint",
+      "outputs": [
+        "{options.outputFile}"
+      ],
+      "options": {
+        "lintFilePatterns": [
+          "sdk/nodejs/**/*.ts"
+        ]
+      }
+    },
+    "test": {
+      "executor": "@nx/jest:jest",
+      "outputs": [
+        "{workspaceRoot}/coverage/sdk/nodejs"
+      ],
+      "options": {
+        "jestConfig": "sdk/nodejs/jest.config.ts",
+        "passWithNoTests": false,
+        "codeCoverage": true
+      },
+      "dependsOn": [
+        {
+          "target": "build"
         },
-        "npm-publish": {
-            "executor": "nx:run-commands",
-            "options": {
-                "command": "../../../scripts/npm-safe-publish.sh \"@devcycle/nodejs-server-sdk\"",
-                "cwd": "dist/sdk/nodejs",
-                "forwardAllArgs": true
-            }
+        {
+          "target": "build:json",
+          "dependencies": true
+        }
+      ]
+    },
+    "build": {
+      "executor": "@nx/js:tsc",
+      "outputs": [
+        "{options.outputPath}"
+      ],
+      "options": {
+        "outputPath": "dist/sdk/nodejs",
+        "tsConfig": "sdk/nodejs/tsconfig.lib.json",
+        "packageJson": "sdk/nodejs/package.json",
+        "buildableProjectDepsInPackageJsonType": "dependencies",
+        "main": "sdk/nodejs/src/index.ts",
+        "assets": [
+          "sdk/nodejs/*.md"
+        ],
+        // External pacakges now need to be defined here, all other pacakges will be compiled into the sdk.
```

Comment:
technically you cant add comments to json files

Reply:
yea I know, but also it works with them... so I thought it would be helpful to try and explain what external packages are (one of the reasons JSON isn't a good format for config files).